src/wazuh_db/wdb_agent.c

```diff
@@ -375,181 +287,201 @@ int wdb_remove_agent_db(int id, const char * name) {
+    int wdb_sock = -1;
+    char *json_string = NULL;
+    cJSON *root = NULL;
+    cJSON *elem = NULL;
+    cJSON *json_name = NULL;
 
     switch (type) {
     case WDB_SYSCHECK:
-        sql = SQL_SELECT_FIM_OFFSET;
+        snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_SELECT_FIM_OFFSET], id_agent);
+        is_FIM = TRUE;
         break;
     case WDB_SYSCHECK_REGISTRY:
-        sql = SQL_SELECT_REG_OFFSET;
+        snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_SELECT_REG_OFFSET],id_agent);
+        is_FIM = FALSE;
         break;
     default:
         return -1;
     }
 
-    if (wdb_open_global() < 0)
+    wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
+ 
+    if(json_string = wstr_chr(wdboutput, ' '), json_string ){
+        *json_string='\0';
+        json_string++;
+    } else{
+        mdebug1("SQLite result has no space. Query: %s",wdbquery);
         return -1;
+    }
 
-    if (wdb_prepare(wdb_global, sql, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
+    if(strcmp(wdboutput,"ok") == 0 && strcmp(json_string,"[]") != 0){
+        root = cJSON_Parse(json_string);    
+        elem = cJSON_GetArrayItem(root, 0);
+        json_name = is_FIM == TRUE ? cJSON_GetObjectItem(elem, "fim_offset") : cJSON_GetObjectItem(elem, "reg_offset");
+        result = json_name->valueint;
+    } else{
+        mdebug1("SQLite Query failed: %s", wdbquery);
         return -1;
     }
 
-    sqlite3_bind_int(stmt, 1, id_agent);
-    result = wdb_step(stmt) == SQLITE_ROW ? sqlite3_column_int64(stmt, 0) : -1;
-    sqlite3_finalize(stmt);
-
     return result;
 }
 
-/* Set the file offset. Returns number of affected rows, or -1 on failure. */
+/* Set the file offset. Returns 1, or -1 on failure. */
 int wdb_set_agent_offset(int id_agent, int type, long offset) {
-    int result;
-    const char *sql;
-    sqlite3_stmt *stmt;
+    int result = 0;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
 
     switch (type) {
     case WDB_SYSCHECK:
-        sql = SQL_UPDATE_FIM_OFFSET;
+        snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_UPDATE_FIM_OFFSET], offset, id_agent);
         break;
     case WDB_SYSCHECK_REGISTRY:
-        sql = SQL_UPDATE_REG_OFFSET;
+        snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_UPDATE_REG_OFFSET], offset, id_agent);
         break;
     default:
         return -1;
     }
 
-    if (wdb_open_global() < 0)
-        return -1;
-
-    if (wdb_prepare(wdb_global, sql, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
-    }
-
-    sqlite3_bind_int64(stmt, 1, offset);
-    sqlite3_bind_int(stmt, 2, id_agent);
-
-    result = wdb_step(stmt) == SQLITE_DONE ? (int)sqlite3_changes(wdb_global) : -1;
-    sqlite3_finalize(stmt);
-
-    return result;
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
```

Comment:
Log information about the result and make use of `OS_SUCCESS` and `OS_INVALID`

Reply:
Done!